cmake/WrongCompilerOptions.cmake

```diff
@@ -0,0 +1,19 @@
+if(APPLE)
+    # It seems that there are some flags added to CXXFLAGS during `conda-build`. It might be a bug
+    # in https://github.com/conda-forge/clang-compiler-activation-feedstock. One way to fix this is
+    # to turn these errors to warnings or silence them (using `-Wno-unused-command-line-argument`
+    # and `-Wno-ignored-optimization-argument`)
+
+    # But we will go another way: we will remove wrong options from CMAKE_CXX_FLAGS
```

Comment:
This is the most tricky part.

I think we have some flags, which are not part of this project and we inherit it during `conda-build`.

This is how current `main` looks like: https://github.com/mamba-org/mamba/actions/runs/11896974442/job/33150163684#step:7:1203

```
x86_64-apple-darwin13.4: warning: -framework CoreFoundation: 'linker' input unused [-Wunused-command-line-argument]
x86_64-apple-darwin13.4: warning: -framework CoreServices: 'linker' input unused [-Wunused-command-line-argument]
x86_64-apple-darwin13.4: warning: -framework Security: 'linker' input unused [-Wunused-command-line-argument]
x86_64-apple-darwin13.4: warning: -framework Kerberos: 'linker' input unused [-Wunused-command-line-argument]
x86_64-apple-darwin13.4: warning: optimization flag '-fno-merge-constants' is not supported [-Wignored-optimization-argument]
```

I think it comes from here `INFO: activate_clangxx_osx-64.sh made the following environmental changes:`:
https://github.com/mamba-org/mamba/actions/runs/11896974442/job/33150163684#step:7:1123

I'm not 100% sure about this and might be completely wrong.
If someone more experienced could take a look at this, it would be great.

Reply:
Trying another approach right now

Reply:
It worked; this one didn't due to some strange linking errors.